testdata/sqllogictests_sqlite/validation.slt

```diff
@@ -1,28 +1,27 @@
 # Validation tests for sqlite external database and external tables
+#
 
-# External database validation
+# Originally these operations failed because GlareDB validated that
+# the table existed when the table was created in the catalog, but
+# these should error at query time: they may fail to exist now but may
+# exist later.
```

Comment:
When you create an external (anything) in (many? most?) sources, they `attempt` to connect to the provider to verify that the resource exists. This is cool, particularly for data sources where the "connect" option is relatively inexpensive, and means that we can catch some errors early (potentially,) but this check doesn't mean that the resource will still exist at query time, and everything could change (of course.) 

Access a non-local sqlite database is comparatively expensive, and because we release the table providers at the end of a query, we'd (in the current formulation) download the entire database just to make sure that it existed. That seems bad, and being lazy here seems like the right answer. 

(as I wrote this, I'm overcome with some doubt about this,)

Reply:
Ok, doubt resolved: 

https://github.com/glaredb/glaredb/blob/tycho/sqlite-from-cloud/crates/sqlexec/src/planner/session_planner.rs#L698-L756

The `SqliteAccess` object is now the thing that downloads the remote files, and all of the `create_external_` operations establish test connections and check that the resource exists. This is probably cheap enough to be fine (though for postgres if you were doing this often DBAs/admins would probably notice connection churn. We could do some generic object store access thing and check that there's exactly one file that matches the location, _but_ files, credentials, paths, can all change between resource creation and first query. 

The [external mongodb table support](https://github.com/glaredb/glaredb/blob/tycho/sqlite-from-cloud/crates/sqlexec/src/planner/session_planner.rs#L581-L597) doesn't create a test connection, but everything else seems to. 

(Interestingly, of course, if you `create-external-table-{gcs,s3,azure}` for a `.sqlite` extension it'll test for the existence of the file and not download it.)